src/components/header/modals/ModalPolkasafe.vue

```
@@ -0,0 +1,394 @@
+          const { data, error } = await c.connectAddress(signatory);
+          if (error) throw Error(error);
+          if (!data.hasOwnProperty('multisigAddresses')) {
+            throw Error('error fetching multisig accounts');
+          }
+
+          //@ts-ignore
+          const filteredMultisigAccounts = data.multisigAddresses.filter((it: MultisigAddress) =>
+            isValidAddressPolkadotAddress(it.address, ASTAR_ADDRESS_PREFIX)
+          );
+          if (filteredMultisigAccounts.length > 0) {
+            const updatedAccountMap = await Promise.all(
+              filteredMultisigAccounts.map(async (it: MultisigAddress) => {
+                const balance = await fetchNativeBalance({
+                  api: $api as ApiPromise,
+                  address: it.address,
+                });
+                return { ...it, balance };
+              }) as MultisigAddress[]
+            );
+            multisigAccounts.value = updatedAccountMap.sort(
+              (a: MultisigAddress, b: MultisigAddress) => Number(b.balance) - Number(a.balance)
+            );
+          }
+        };
+
+        const handleInitializePolkasafe = async (): Promise<void> => {
+          const client = new Polkasafe();
+          await client.connect('astar', signatory, injector);
+          container.addConstant<Polkasafe>(Symbols.PolkasafeClient, client);
+          await setMultisigAccounts(client);
+        };
+
+        try {
+          const polkasafeClient = container.get<Polkasafe>(Symbols.PolkasafeClient);
+          const isSigned = polkasafeClient && polkasafeClient.address === signatory;
+          if (isSigned) {
+            await setMultisigAccounts(polkasafeClient);
+          } else {
+            await handleInitializePolkasafe();
+          }
+        } catch (error) {
+          // Memo: polkasafeClient has not been initialized
+          await handleInitializePolkasafe();
+        }
+      } catch (error) {
+        console.error(error);
```

Don't swallow, handle in line 346 and show error toast

---

Fixed in [this](https://github.com/AstarNetwork/astar-apps/pull/834/commits/6d8f31f2645b3e46060408ae8ca42e1fa7ba542d) commit.